specification/mongocluster/DocumentDB.MongoCluster.Management/MongoCluster.tsp

```diff
@@ -209,3 +243,109 @@ union PublicNetworkAccess {
   "Disabled",
 }
+
+/** Preview features that can be enabled on a mongo cluster. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+union PreviewFeature {
+  string,
+
+  /** Enables geo replicas preview feature. The feature must be set at create-time on new cluster to enable linking a geo-replica cluster to it. */
+  "GeoReplicas",
+}
+
+/** Parameters used for replica operations. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+model MongoClusterReplicaParameters {
+  /** The id of the replication source cluster. */
+  sourceResourceId: string;
+
+  /** The location of the source cluster */
+  sourceLocation: string;
+}
+
+/** Promote replica request properties. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+model PromoteReplicaRequest {
+  /** The promote option to apply to the operation. */
+  promoteOption: PromoteOption;
+
+  /** The mode to apply to the promote operation. Value is optional and default value is 'Switchover'. */
+  mode?: PromoteMode;
+}
+
+/** Promote request results. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+model PromoteReplicaResult {}
+
+/** The option to apply to a promote operation. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+union PromoteOption {
+  string,
+
+  /** Promote option forces the promotion without waiting for the replica to be caught up to the primary. This can result in data-loss so should only be used during disaster recovery scenarios. */
+  "Forced",
+}
+
+/** The mode to apply to a promote operation. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+union PromoteMode {
+  string,
+
+  /** Promotion will switch the current replica cluster to the primary role and the original primary will be switched to a replica role, maintaining the replication link. */
+  "Switchover",
+}
+
+/** Replica properties of the mongo cluster. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+model ReplicationProperties {
+  /** The resource id the source cluster for the replica cluster. */
+  @visibility("read")
+  sourceResourceId?: string;
+
+  /** The replication role of the cluster */
+  @visibility("read")
+  role?: ReplicationRole;
+
+  /** The replication link state of the replica cluster. */
+  @visibility("read")
+  replicationState?: ReplicationState;
+}
+
+/** Replication role of the mongo cluster. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+union ReplicationRole {
+  string,
+
+  /** The cluster is a primary replica. */
+  "Primary",
+
+  /** The cluster is a local asynchronous replica. */
+  "AsyncReplica",
+
+  /** The cluster is a geo-asynchronous replica. */
+  "GeoAsyncReplica",
+}
+
+/** The state of the replication link between the replica and source cluster. */
+@added(Microsoft.DocumentDB.Versions.v2024_06_01_preview)
+union ReplicationState {
+  string,
+
+  /** Replication link is active. */
+  "Active",
+
+  /** Replication link is catching-up. */
+  "Catchup",
```

Comment:
Can you add descriptions to clarify what these transition states indicate about what the service is doing? In particular, what is the difference between Catchup, Provisioning, Updating, and Reconfiguring?